### PR TITLE
Add SRP batcher support

### PR DESCRIPTION
**Pull request by Kink3d**
### Purpose of this PR
This PR adds support for SRP batcher in shaders generated by shader graph. It does this by defining `isBatchable` per shader property type then filtering them based on this on the `PropertyCollector`. Batchable properties are then inserted in the `UnityPerMaterial` c buffer.

---
### Release Notes
See changelog

---
### Testing status
**Katana Tests**: Running...

**Manual Tests**: @zadi0648 / @spaztronaut 

**Automated Tests**: None

Any test projects to go with this to help reviewers? None

---
### Overall Product Risks
**Technical Risk**: None, Low, Medium, High?
Low - Minimal changes

**Halo Effect**: None, Low, Medium, High?
Low - Affects all properties, but minimal code footprint

---
### Comments to reviewers
None

### CODE REVIEW

File: com.unity.shadergraph/Editor/Data/Graphs/Matrix2ShaderProperty.cs [hunk 1]
```diff
@@ -17,6 +17,11 @@ public override PropertyType propertyType
             get { return PropertyType.Matrix2; }
         }
 
+        public override bool isBatchable
+        {
+            get { return false; }
```
Comment (cdxntchou) [inline com.unity.shadergraph/Editor/Data/Graphs/Matrix2ShaderProperty.cs]: Why are Vectors batchable but not Matrices?

Comment (Kink3d) [reply com.unity.shadergraph/Editor/Data/Graphs/Matrix2ShaderProperty.cs]: Good point, I guess i should make the matrices batchable, thanks

**Comment by arnaudcarre** [comment]
Hi! don't know if the term "is batchable" will be public or not. In this context, I guess "this var is a material parameter" is better approach. In another way, imagine a standard unsity shader. If any property is accessible from the material inspector pannel, then this var is a material parameter and should land in "UnityPerMaterial" constant buffer. ( as opposed to builtin vars or global var )
Feel free to discuss about it on slack if needed ( @arnaud )

File: com.unity.shadergraph/Editor/Data/Graphs/GradientShaderProperty.cs [hunk 2]
```diff
@@ -114,6 +114,11 @@ public override Vector4 defaultValue
             get { return new Vector4(); }
         }
 
+        public override bool isBatchable
+        {
+            get { return false; }
```
[the remark this answers is not in the document]
Comment (Kink3d) [reply com.unity.shadergraph/Editor/Data/Graphs/GradientShaderProperty.cs]: They could and prbably should, but that involves a gradient refactor. Will happen in a different PR.

**Comment by Kink3d** [comment]
Tests green (against 2018.3/release)

**Pull request merged**